### PR TITLE
feat: Add Registry/Config Access token capability via Secret Provider

### PR DESCRIPTION
**Pull request by lenny-intel**
closes 

BREAKING CHANGE: When run with the secure Edgex Stack now need to have the SecretStore configured, a Vault token created and run with EDGEX_SECURITY_SECRET_STORE=true.
BREAKING CHANGE: Service key changed to `app-<profile name>`

Signed-off-by: lenny <leonard.goodell@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Registry/Config access tokens capability supported via files

Issue Number:  #222

## What is the new behavior?
Registry/Config access tokens capability now supported via SecretProvider
Non-standard mixed case service key name change to simplified lower case `app-<profile name>`
Removed previous file based access token config settings

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

BREAKING CHANGE: When run with the secure Edgex Stack now need to have the SecretStore configured, a Vault token created and run with EDGEX_SECURITY_SECRET_STORE=true.
BREAKING CHANGE: Service key changed to `app-<profile name>`

## Are there any new imports or modules? If so, what are they used for and why?
no

## Other information

### CODE REVIEW

**Comment by lenny-intel** [comment]
The TAF tests will need to be update for the change in service key names. Thus this PR is in draft mode until those changes can be made.

File: res/rules-engine/configuration.toml [hunk 1]
```diff
@@ -63,3 +63,20 @@ Type = "zero"
   Port = 5566
   Protocol = "tcp"
```
Comment (jim-wang-intel) [inline res/rules-engine/configuration.toml:65]: need to remove `ConfigAccessTokenFile ` and `AccessTokenFile ` from the above config

File: res/functional-tests/configuration.toml [hunk 2]
```diff
@@ -88,3 +88,21 @@ AccessTokenFile = '/tmp/edgex/secrets/AppService-functional-tests/consul-token'
 [Trigger]
```
Comment (jim-wang-intel) [inline res/functional-tests/configuration.toml:88]: need to remove `ConfigAccessTokenFile ` and `AccessTokenFile ` from the above config.

File: res/http-export/configuration.toml [hunk 3]
```diff
@@ -97,18 +97,18 @@ Port = 6379
 Timeout = "30s"
```
Comment (jim-wang-intel) [inline res/http-export/configuration.toml:97]: need to remove `ConfigAccessTokenFile ` and `AccessTokenFile ` from the above config.

File: res/mqtt-export/configuration.toml [hunk 4]
```diff
@@ -103,18 +103,18 @@ Port = 6379
 Timeout = "30s"
```
Comment (jim-wang-intel) [inline res/mqtt-export/configuration.toml:103]: need to remove `ConfigAccessTokenFile ` and `AccessTokenFile ` from the above config.

File: res/push-to-core/configuration.toml [hunk 5]
```diff
@@ -35,3 +35,21 @@ AccessTokenFile = '/tmp/edgex/secrets/AppService-push-to-core/consul-token' # ig
```
Comment (jim-wang-intel) [inline res/push-to-core/configuration.toml:35]: need to remove `ConfigAccessTokenFile ` and `AccessTokenFile ` from the above config.

File: res/rules-engine/configuration.toml [hunk 6]
```diff
@@ -63,3 +63,20 @@ Type = "zero"
...
+#   - SecretStore_Host: edgex-vault
+[SecretStore]
+Type = 'vault'
+Host = 'localhost'
+Port = 8200
+Path = '/v1/secret/edgex/app-rules-engine/'
+Protocol = 'http'
+RootCaCertPath = ''
+ServerName = ''
+TokenFile = '/tmp/edgex/secrets/app-rules-engine/secrets-token.json'
```
Comment (jim-wang-intel) [inline res/rules-engine/configuration.toml:78]: so we want `app-rules-engine` as the role name in security-bootstrapper side, right?

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
some file-based leftover configuration to be removed

File: res/rules-engine/configuration.toml (hunk 6 above)
Comment (lenny-intel) [reply res/rules-engine/configuration.toml:78]: Correct

File: res/push-to-core/configuration.toml (hunk 5 above)
Comment (lenny-intel) [reply res/push-to-core/configuration.toml:35]: Good catch. Thought I had got all those...

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**